src/test/regress/sql/failure_insert_select_via_coordinator.sql

```diff
@@ -0,0 +1,97 @@
+
+-- kill coordinator pull query
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").kill()');
+INSERT INTO events_summary SELECT event_id, event_type, count(*) FROM events_table GROUP BY 1,2;
+
+-- kill data push
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY coordinator_insert_select").kill()');
+INSERT INTO events_summary SELECT event_id, event_type, count(*) FROM events_table GROUP BY 1,2;
+
+-- cancel coordinator pull query
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").cancel(' || :pid || ')');
+INSERT INTO events_summary SELECT event_id, event_type, count(*) FROM events_table GROUP BY 1,2;
+
+-- cancel data push
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY coordinator_insert_select").cancel(' || :pid || ')');
+INSERT INTO events_summary SELECT event_id, event_type, count(*) FROM events_table GROUP BY 1,2;
+
+--verify nothing is modified
+SELECT citus.mitmproxy('conn.allow()');
+SELECT count(*) FROM events_summary;
+
+-- insert into reference table from a distributed table
+-- kill coordinator pull query
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").kill()');
+INSERT INTO events_reference SELECT event_type, count(*) FROM events_table GROUP BY 1;
+
+-- kill data push
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY coordinator_insert_select").kill()');
+INSERT INTO events_reference SELECT event_type, count(*) FROM events_table GROUP BY 1;
+
+-- cancel coordinator pull query
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").cancel(' || :pid || ')');
+INSERT INTO events_reference SELECT event_type, count(*) FROM events_table GROUP BY 1;
+
+-- cancel data push
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY coordinator_insert_select").cancel(' || :pid || ')');
+INSERT INTO events_reference SELECT event_type, count(*) FROM events_table GROUP BY 1;
+
+--verify nothing is modified
+SELECT citus.mitmproxy('conn.allow()');
+SELECT count(*) FROM events_reference;
+
+-- insert/select from reference table to distributed
+
+-- fill up reference table first
+INSERT INTO events_reference SELECT event_type, count(*) FROM events_table GROUP BY 1;
+
+-- kill coordinator pull query
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").kill()');
+INSERT INTO events_reference_distributed SELECT * FROM events_reference;
+
+-- kill data push
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY coordinator_insert_select").kill()');
+INSERT INTO events_reference_distributed SELECT * FROM events_reference;
+
+-- cancel coordinator pull query
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY").cancel(' || :pid || ')');
+INSERT INTO events_reference_distributed SELECT * FROM events_reference;
+
+-- cancel data push
+SELECT citus.mitmproxy('conn.onQuery(query="^COPY coordinator_insert_select").cancel(' || :pid || ')');
+INSERT INTO events_reference_distributed SELECT * FROM events_reference;
+
+--verify nothing is modified
+SELECT citus.mitmproxy('conn.allow()');
+SELECT count(*) FROM events_reference_distributed;
```

Comment:
Same; I would suggest to have this test after every interrupted transaction. It will give better guidance where a bug might have been introduced.